soc/arm/nuvoton/m48x/soc.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2020 Linumiz
+ * Author: Saravanan Sekar <saravanan@linumiz.com>
+ */
+
+#include <init.h>
+#include <kernel.h>
+
+void z_platform_init(void)
+{
+	SYS_UnlockReg();
+
+	/* system clock init */
+	SystemInit();
+
+	/* Enable HXT clock (external XTAL 12MHz) */
+	CLK_EnableXtalRC(CLK_PWRCTL_HXTEN_Msk);
+
+	/* Wait for HXT clock ready */
+	CLK_WaitClockReady(CLK_STATUS_HXTSTB_Msk);
+
+	/* Set core clock as PLL_FOUT source */
+	CLK_SetCoreClock(FREQ_192MHZ);
+
+	/* Set both PCLK0 and PCLK1 as HCLK/2 */
+	CLK->PCLKDIV = (CLK_PCLKDIV_APB0DIV_DIV2 | CLK_PCLKDIV_APB1DIV_DIV2);
+
+	SystemCoreClockUpdate();
+
+	SYS_LockReg();
```

Comment:
Note that these functions are executed before the bss section is zeroed and the data sections are copied to the RAM.

Are you completely certain that none of these functions and the functions called by them make use of the pre-loaded data section contents or rely on the initialised bss section? If not, these should be called through `SYS_INIT` after `prep_c`.

Reply:
@stephanosio thanks and I aware of the hook z_platform_init is called before C init. non of these functions uses global/static variables and all of these are direct register address accessing.